### PR TITLE
chore: Rewrite `terraform_validate` to `common::per_dir_hook`

### PR DESCRIPTION
**Pull request by MaxymVlasov**
That's one of many PRs to the `pre-release` branch, each one will depend on the previous one.
That is the way to include all that info in one release...

Also, close #402.

### CODE REVIEW

File: hooks/terraform_validate.sh [hunk 1]
```diff
@@ -17,14 +17,26 @@ function main {
...
-  terraform_validate_
+
+  # Export provided env var K/V pairs to environment
+  local var var_name var_value
+  for var in "${ENVS[@]}"; do
+    var_name="${var%%=*}"
+    var_value="${var#*=}"
+    # shellcheck disable=SC2086
+    export $var_name="$var_value"
+  done
```
Comment (MaxymVlasov) [inline hooks/terraform_validate.sh:28]: That's will be moved to separate function later

File: hooks/terraform_validate.sh [hunk 2]
```diff
@@ -45,6 +57,11 @@ function parse_cmdline_ {
         ARGS+=("$1")
         shift
         ;;
+      -h | --hook-config)
```
Comment (MaxymVlasov) [inline hooks/terraform_validate.sh:60]: `parse_cmdline_` will be merged in `common::parse_cmdline` later

File: hooks/terraform_validate.sh [hunk 3]
```diff
@@ -65,88 +82,53 @@ function parse_cmdline_ {
...
+    init_output=$(terraform init -backend=false "${INIT_ARGS[@]}" 2>&1)
+    exit_code=$?
+
+    if [ $exit_code -ne 0 ]; then
+      common::colorify "yellow" "'terraform init' failed, 'terraform validate' skipped: $dir_path"
+      echo "$init_output"
+      return $exit_code
     fi
-  done
+  fi
```
Comment (MaxymVlasov) [inline hooks/terraform_validate.sh:118]: Will be moved to `common::terraform_init` function

**Review by MaxymVlasov — COMMENTED** [review]
Leave some notes about future improvements, to avoid extra attention to them in this PR

File: hooks/terraform_validate.sh [hunk 4]
```diff
@@ -65,88 +82,53 @@ function parse_cmdline_ {
...
+  local init_output
+  local validate_output
+
+  if [ ! -d .terraform ]; then
+    init_output=$(terraform init -backend=false "${INIT_ARGS[@]}" 2>&1)
+    exit_code=$?
+
+    if [ $exit_code -ne 0 ]; then
+      common::colorify "yellow" "'terraform init' failed, 'terraform validate' skipped: $dir_path"
+      echo "$init_output"
```
Comment (yermulnik) [inline hooks/terraform_validate.sh:115]: should the `echo` use `-e` opt to interpret escape sequences or `terraform init` should use `-no-color`?

**Review by yermulnik — APPROVED** [review]

File: hooks/terraform_validate.sh (hunk 4 above)
Comment (MaxymVlasov) [reply hooks/terraform_validate.sh:115]: Ok, will replace `echo` with  `common::colorify`

Comment (yermulnik) [reply hooks/terraform_validate.sh]: The other `echo` used to output `terraform validate` then needs this change too.
Though... I'd better keep native TF output colorize specs to keep it usual and recognized by users.

Comment (yermulnik) [reply hooks/terraform_validate.sh]: Especially given that there's not `-no-color` passed to terraform, its escape sequences may (and probably will) get mixed up with sequences used by `common::colorify` and the output may become vomit of all the colours of the rainbow 🤣

Comment (MaxymVlasov) [reply hooks/terraform_validate.sh]: Just tested - terraform ignore any color instructions specified to pre-commit or by `common::colorify`

Will be done in another PR (need to disable that across all hooks)

File: hooks/terraform_validate.sh (hunk 3 above)
Comment (MaxymVlasov) [reply hooks/terraform_validate.sh:118]: Done in #406

File: hooks/terraform_validate.sh (hunk 1 above)
Comment (MaxymVlasov) [reply hooks/terraform_validate.sh:28]: Done in #408

File: hooks/terraform_validate.sh (hunk 2 above)
Comment (MaxymVlasov) [reply hooks/terraform_validate.sh:60]: Done in #408

File: hooks/terraform_validate.sh (hunk 4 above)
Comment (MaxymVlasov) [reply hooks/terraform_validate.sh:115]: Done in #409

**Pull request merged**